tajo-core-tests/src/test/java/org/apache/tajo/engine/planner/TestPartitionedTableRewriter.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.tajo.engine.planner;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.tajo.LocalTajoTestingUtility;
+import org.apache.tajo.OverridableConf;
+import org.apache.tajo.QueryTestCaseBase;
+import org.apache.tajo.algebra.Expr;
+import org.apache.tajo.catalog.CatalogUtil;
+import org.apache.tajo.catalog.TableDesc;
+import org.apache.tajo.engine.query.QueryContext;
+import org.apache.tajo.plan.LogicalPlan;
+import org.apache.tajo.plan.logical.*;
+import org.apache.tajo.plan.rewrite.rules.PartitionedTableRewriter;
+import org.apache.tajo.util.CommonTestingUtil;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class TestPartitionedTableRewriter extends QueryTestCaseBase {
```

Comment:
You must add a test to check that findFilteredPartitionPaths() works well after a manual partition directory creation. This test is related to the following bug.

```
# I manually added a directory 'n_comment=this is a directory to test partition' under nation_part directory.
default> \dfs -ls /tajo/warehouse/default/nation_part
Found 26 items
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment= haggle. carefully final deposits detect slyly agai
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment= pending excuses haggle furiously deposits. pending, express pinto beans wake fluffily past t
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment= requests against the platelets use never according to the quickly regular pint
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment= slyly express asymptotes. regular deposits haggle slyly. carefully ironic hockey players sleep blithely. carefull
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=al foxes promise slyly according to the regular accounts. bold requests alon
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=c dependencies. furiously express notornis sleep slyly regular accounts. ideas sleep. depos
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=eans boost carefully special requests. accounts are. carefull
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=eas hang ironic, silent packages. slyly regular packages are furiously over the tithes. fluffily bold
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=efully alongside of the slyly final dependencies. 
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=hely enticingly express accounts. even, final 
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ic deposits are blithely about the carefully regular pa
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=l platelets. regular accounts x-ray%3A unusual, regular acco
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=nic deposits boost atop the quickly final requests%3F quickly regula
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ously. final, express gifts cajole a
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=platelets. blithely pending dependencies use fluffily across the even pinto beans. carefully silent accoun
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=refully final requests. regular, ironi
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=rns. blithely bold courts among the closely regular packages use furiously bold platelets%3F
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=s. ironic, unusual asymptotes wake blithely r
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ss excuses cajole slyly across the packages. deposits print aroun
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 17:45 /tajo/warehouse/default/nation_part/n_comment=this is a directory to test partition.
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ts. silent requests haggle. closely express packages sleep across the blithely
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ular asymptotes are about the furious multipliers. express dependencies nag above the ironically ironic account
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=ven packages wake quickly. regu
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=y above the carefully unusual theodolites. final dugouts are quickly across the furiously regular d
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=y alongside of the pending deposits. carefully special packages are about the ironic forges. slyly special 
drwxr-xr-x   - jihoon supergroup          0 2015-10-15 13:03 /tajo/warehouse/default/nation_part/n_comment=y final packages. slow foxes cajole quickly. quickly silent platelets breach ironic accounts. unusual pinto be
default> select n_comment from nation_part where substr(n_comment, 0, 8) = 'this is';
Progress: 100%, response time: 2.031 sec
n_comment
-------------------------------
(0 rows, 2.031 sec, 0 B selected)
```